File: old/meeting.rs
use crate::tools::Days;
use std::cmp::Ordering;
use time::Time;

/// The `Meeting` type holds individual meeting information that is retrieved from the SQLite database.
#[derive(Clone, Debug, Eq, Ord, PartialEq)]
pub struct Meeting {
    pub day: Days,
    pub hours: [Time; 2],
    pub instructor: String,
    pub location: String,
    pub meeting_id: String,
    pub meeting_type: String,
}

impl PartialOrd for Meeting {
    fn partial_cmp(&self, other: &Meeting) -> Option<Ordering> {
        Some(self.hours[0].cmp(&other.hours[0]))
    }
}

File: hourglass/meeting.py
from hourglass.period import DateTime


class Meeting:
    """
    Store meeting information for Course.

    Parameters
    ----------
    date : DateTime
        Date of the meeting.
    instructor : str
        Name of the instructor.
    location : str
        Location of the meeting.
    meeting_id : str
        Meeting ID from the database.
    meeting_type : str
        Type of meeting.
    """

    def __init__(
        self,
        date: DateTime = DateTime(),
        instructor: str = "",
        location: str = "",
        meeting_id: str = "",
        meeting_type: str = "",
    ):
        self.date = date
        self.instructor = instructor
        self.location = location
        self.meeting_id = meeting_id
        self.meeting_type = meeting_type
